src/passes/Flatten.cpp

```diff
@@ -192,6 +193,40 @@ struct Flatten
         loop->finalize();
         replaceCurrent(rep);
 
+      } else if (auto* tryy = curr->dynCast<Try>()) {
+        // remove a try value
+        Expression* rep = tryy;
+        auto* originalBody = tryy->body;
+        std::vector<Expression*> originalCatchBodies(tryy->catchBodies.begin(),
+                                                     tryy->catchBodies.end());
+        auto type = tryy->type;
+        Expression* prelude = nullptr;
+        if (type.isConcrete()) {
+          Index temp = builder.addVar(getFunction(), type);
+          if (tryy->body->type.isConcrete()) {
```

Comment:
I think this if just skips adding a set if the body is unreachable, as a minor optimization? I don't think it's needed IIUC. And the catch bodies do not do this check which makes it a little surprising to see it here.

Reply:
I copy-pasted `if`'s code here again: https://github.com/WebAssembly/binaryen/blob/3c1c8ae829d0026024de55aa65dc3d49bd8fd37a/src/passes/Flatten.cpp#L152-L159

I tried to remove these `if (iff->ifTrue->type.isConcrete())` and `if (iff->ifFalse && iff->ifFalse->type.isConcrete())` to see if these also can be removed, but if I do that these test fail:
```
Failed Tests (2):
  Binaryen lit tests :: passes/flatten_simplify-locals-nonesting_souperify-single-use_enable-threads.wast
  Binaryen lit tests :: passes/flatten_simplify-locals-nonesting_souperify_enable-threads.wast
```

I haven't checked why they fail (and I have to run for today), but I guess there can be cases these are necessary...? If that's the case I think I should also check the catch bodies too. I'll do that for the moment, and if they turn out not to be necessary, I will remove them.

Reply:
I see, thanks. In that case let's check the catch bodies too, that seems most consistent with the rest of the code.

Reply:
I think it's OK to remove these body checks (both in `if` and `try`); the reason those tests failed was not because they crashed, but because the filecheck results were different. As you said, this does a small optimization that prevents this form:
```wasm
(local.set $x
  (unreachable)
)
```

But I think it's OK to leave them (both in `if` and `try`) anyway; while `Flatten` doesn't try to do any optimizations itself, it wouldn't hurt, and maybe the output will be a little easier to read...

Reply:
sgtm, this simple pattern doesn't add much code complexity and it does make the output smaller and simpler.